src/scheduler/MpiWorld.cpp

```diff
@@ -113,13 +185,37 @@ void MpiWorld::create(const faabric::Message& call, int newId, int newSize)
 
 void MpiWorld::destroy()
 {
-    // Destroy once per host
-    if (!isDestroyed.test_and_set()) {
-        shutdownThreadPool();
+    // Destroy once per thread the rank-specific data structures
+    // Remote message endpoints
+    if (mpiMessageEndpoints.size() > 0) {
```

Comment:
I think this and other `foo.size() > 0` is a clang-tidy warning about using `empty` when checking for emptiness as it's a bit more readable (i.e. `if(!mpiMessageEndpoints.size().empty())`)